### PR TITLE
Change default cache directory

### PR DESCRIPTION
**Pull request by mistic**
This PR closes https://github.com/webpack-contrib/cache-loader/issues/62
It is pretty simple and only change the default cache directory.

/CC @evilebottnawi

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -19,7 +19,7 @@ const schema = require('./options.json');
 
 const defaults = {
   cacheContext: '',
-  cacheDirectory: path.resolve('.cache-loader'),
+  cacheDirectory: path.resolve('./node_modules/.cache/.cache-loader'),
```
Comment (evilebottnawi) [inline src/index.js]: Can be broken when you use lerna and other nonstandard project structure, to avoid this we need use `find-cache-dir` package

Comment (mistic) [reply src/index.js]: @evilebottnawi you're right, just forgot about this! Thanks for the feedback 👍

File: src/index.js [hunk 2]
```diff
@@ -19,7 +21,7 @@ const schema = require('./options.json');
 
 const defaults = {
   cacheContext: '',
-  cacheDirectory: path.resolve('./node_modules/.cache/.cache-loader'),
+  cacheDirectory: findCacheDir({ name: 'cache-loader' }) || os.tmpdir(),
```
Comment (evilebottnawi) [inline src/index.js]: no need `os.tmpdir`, `find-cache-dir` already has fallback when directory was not found

Comment (mistic) [reply src/index.js]: I think we are good here as reading the documentation of `find-cache-dir` doesn't provide any hint about a fallback. Are u okay with it @evilebottnawi ?

Comment (evilebottnawi) [reply src/index.js]: @mistic will be great send a PR about this in `find-cache-dir` too, but not related to this PR

**Comment by evilebottnawi** [comment]
I am afraid it is breaking change

File: src/index.js (hunk 2 above)
Comment (mistic) [reply src/index.js]: I'll do it @evilebottnawi  😃

**Comment by mistic** [comment]
@evilebottnawi what do you suggest? waiting for the next major release?

**Comment by evilebottnawi** [comment]
@mistic i think yes, anyway webpack@5 have built-in cache module, so i suggest see where he stored cache and send a PR using `node_modules/.cache/webpack` by default. Can you take care?

**Comment by mistic** [comment]
@evilebottnawi I've opened the PR into the webpack repo https://github.com/webpack/webpack/pull/8933

Do you think we can now merge this one here?

**Comment by evilebottnawi** [comment]
@mistic no need issue :+1: We should bump major version after merge, maybe exists other issue with breaking changes?

**Comment by mistic** [comment]
@evilebottnawi let me try and think a little bit!

One thing I'll do is put up another PR in order to fix the build 😄

**Comment by evilebottnawi** [comment]
Let's rebase this PR to avoid CI problems

**Comment by evilebottnawi** [comment]
@mistic ping me if all done for major release and we do this

**Comment by mistic** [comment]
@evilebottnawi I can't think in anything more for the moment that could be a breaking change

**Pull request merged**